sklearn/neighbors/tests/test_graph.py

```diff
@@ -81,6 +84,52 @@ def test_explicit_diagonal():
     assert not _has_explicit_diagonal(X2t)
 
 
+@pytest.mark.parametrize(
+    "params, expected_exception, msg",
+    [
+        (
+            {"p": 0},
+            ValueError,
+            "The 'p' parameter of kneighbors_graph must be a float in the range",
+        ),
+        (
+            {"p": 0.5},
+            UserWarning,
+            "Mind that for 0 < p < 1, Minkowski metrics are not distance metrics",
+        ),
+        (
+            {"p": 1},
+            None,
+            "No exception or warning expected",
+        ),
+        (
+            {"p": 5.7},
+            None,
+            "No exception or warning expected",
+        ),
+        (
+            {"p": float("inf")},
+            None,
+            "No exception or warning expected",
+        ),
+        (
+            {"p": None},
+            TypeError,
+            "'<' not supported between instances of 'NoneType' and 'int'",
+        ),
```

Comment:
p can be None (but actually it can be anything) since it's ignored when p is passed in `metric_params`. It's possible that people set p=None when they pass it in metric_params to be explicit, we do that in some tests iirc. So to not break existing code, let's keep None a valid option.

I wouldn't check for this error as it's not how it should be used.
```suggestion
```